va/va.go

```diff
@@ -432,9 +434,13 @@ func (va *ValidationAuthorityImpl) tryGetTLSSNICerts(ctx context.Context, identi
 		va.stats.Inc("IPv4Fallback", 1)
 	}
 
-	// This shouldn't happen, but be defensive about it anyway
+	// If there are no v4 addresses then return an error about there being no
+	// usable addresses found. We don't say "no IP addresses found" here because
+	// we may have tried an IPv6 address before this point, had it fail, and then
+	// found no fallbacks.
 	if len(v4) < 1 {
-		return nil, validationRecords, probs.Malformed("No available addresses for getTLSSNICerts to dial")
+		return nil, validationRecords, probs.Malformed(
+			fmt.Sprintf("no working IP addresses found for %q", identifier.Value))
```

Comment:
If the consensus is that this is confusing I think the best alternative is to have two error messages, one used when `len(v4) < 1 && no IPv6 was tried`, one for `len(v4) < 1 && IPv6 was tried`.